vividus-plugin-mobitru/src/main/java/org/vividus/mobitru/client/MobitruFacadeImpl.java

```diff
@@ -66,6 +67,20 @@ public MobitruFacadeImpl(MobitruClient mobitruClient)
     @Override
     public String takeDevice(DesiredCapabilities desiredCapabilities) throws MobitruOperationException
     {
+        Map<String, Object> capabilities = desiredCapabilities.asMap();
+        //use different API in case if udid is provided in capabilities
+        //it's required in some cases like if the device is already taken
+        if (!isSearchForDevice(desiredCapabilities)
+                &&
+                (capabilities.containsKey(APPIUM_UDID) || capabilities.containsKey(UDID)))
```

Comment:
why not to move it else-if statement at line 89 so we don't need to potentially invoke isSearchForDevice twice?

Reply:
in order to avoid overcomplicated logic with multiple if else blocks, where it's easy to make a mistake

Reply:
one if-else doesn't look overcomplicated :)

Reply:
who may guarantee that there will be only one here forever?